libraries/ArduinoOTA/ArduinoOTA.cpp

```diff
@@ -146,26 +146,26 @@ int ArduinoOTAClass::parseInt(){
   uint8_t index;
   char value;
   while(_udp_ota->peek() == ' ') _udp_ota->read();
-  for(index = 0; index < sizeof(data); ++index){
+  for(index = 0; index < sizeof(data); index++){
     value = _udp_ota->peek();
     if(value < '0' || value > '9'){
```

Comment:
This if statement needs a check against peek error return: if(value < 0 || everythingelse)